File: face_recognition_app.py
from flask import Flask, render_template, request, jsonify, Response

from face_recognition import start_face_recognition as recognize
from stop_face_recognition import stop_detection
from multiprocessing import Process
from concurrent.futures import ThreadPoolExecutor

app = Flask(__name__)
pool = ThreadPoolExecutor(3)

@app.route('/devcon/face_recognition/start/<profile_id>', methods=['POST'])
def start_face_recognition(profile_id):
    print("Starting face recognition for %s " % profile_id)
    try: 
        #face_app_process = Process(target=recognize, args=(profile_id, True))
        #face_app_process.start()
        #face_app_process.join()
        #pool.submit(recognize, (profile_id, True))
        recognize(profile_id, True)
        return Response('{"response": "Face recognition started"}', status = 200, mimetype='application/json')
    except Exception as err:
        print("Something failed: Druid connection failed")
        print(err)
        raise

@app.route('/devcon/face_recognition/stop/<profile_id>', methods=['POST'])
def stop_face_recognition(profile_id):
    print("Stopping face recognition for %s " % profile_id)
    try: 
        stop_detection(profile_id)
    except Exception as err:
        print("Something failed: Druid connection failed")
        print(err)
        raise

if __name__ == '__main__':
    app.run(debug=True)

File: face_recognition_app.js
// import express JS module into app 
// and creates its variable. 

var express = require('express'); 
var bodyParser = require('body-parser')
var app = express(); 
app.use(bodyParser.urlencoded({ extended: false }));
app.use(bodyParser.json());
  
app.listen(5000, function() { 
    console.log('server running on port 5000'); 
} ) 

app.post('/devcon/face_recognition/start', startFaceRecognition); 
  
function startFaceRecognition(req, res) { 
    var spawn = require("child_process").spawn;  
    var process = spawn('python',["/Users/anand/Documents/code/FaceRecognition/face_recognition.py", 
                            req.query.profile_id] ); 
  
    res.send('{"response": "Successfully started face recognition"'); 
    
    /*
    process.stdout.on('data', function(data) { 
        res.send(data.toString()); 
    }); 
    */
} 

app.post('/devcon/face_recognition/stop', stopFaceRecognition); 
  
function stopFaceRecognition(req, res) { 
    var spawn = require("child_process").spawn;  
    var process = spawn('python',["/Users/anand/Documents/code/FaceRecognition/stop_face_recognition.py", 
                            req.query.profile_id] ); 
  
    res.send('{"response": "Successfully started face recognition"'); 
    
    /*
    process.stdout.on('data', function(data) { 
        res.send(data.toString()); 
    }); 
    */
} 